### PR TITLE
160 no duplicate nodes

### PR DESCRIPTION
**Pull request by kronken**

### CODE REVIEW

File: src/components/sidebar/RouteShow.tsx [hunk 1]
```diff
@@ -29,7 +30,7 @@ class RouteShow extends React.Component<IRouteShowProps> {
     }
 
     private closeRoute() {
-        this.props.routeStore!.removeFromRoutes(this.props.route.routeId);
+        RouteAndStopHelper.removeRoute(this.props.route.routeId);
```
[the remark this answers is not in the document]
Comment (neodyymi) [reply src/components/sidebar/RouteShow.tsx]: As discussed, add a comment TODO move functionality to target url

File: src/storeHelpers/routeAndStopHelper.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import RouteStore from '../stores/routeStore';
```
Comment (neodyymi) [inline src/storeHelpers/routeAndStopHelper.ts]: singletons should begin with lowercase

File: src/factories/routePathFactory.ts [hunk 3]
```diff
@@ -19,26 +25,34 @@ class RoutePathFactory {
...
         const coordinates = JSON.parse(suunta.geojson).coordinates;
         const positions = coordinates.map((coor: [number, number]) => [coor[1], coor[0]]);
 
-        return {
+        const routePath : IRoutePath = {
             routeId,
-            nodes,
             positions,
-            geoJson: JSON.parse(suunta.geojson),
+            routePathLinks: routePathLinks.map(res => res.link),
```
Comment (neodyymi) [inline src/factories/routePathFactory.ts]: One of these should probably be named different. Now they seem like the same thing when they actually aren't.

File: src/factories/routePathFactory.ts [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
+export interface IRoutePathResult {
+    routePath: IRoutePath;
+    nodes: INode[];
+}
 
 class RoutePathFactory {
     // suunta to IRoutePath
     public static createRoutePath = (
         routeId: string,
         suunta: any,
```
Comment (neodyymi) [inline src/factories/routePathFactory.ts]: Not necessarily for this PR, but should this be something else than 'any'

File: src/factories/routePathLinkFactory.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { IRoutePathLink, INode } from '../models';
+import NodeFactory from './nodeFactory';
+
+export interface IRoutePathLinkResult {
+    link: IRoutePathLink;
+    nodes: INode[];
+}
+
+class RoutePathLinkFactory {
+    public static createRoutePathLink = (node: any): IRoutePathLinkResult => {
```
Comment (neodyymi) [inline src/factories/routePathLinkFactory.ts]: How is this a node?

File: src/factories/routePathFactory.ts [hunk 6]
```diff
@@ -19,26 +25,34 @@ class RoutePathFactory {
             ].join('-'),
         ).toString();
 
-        const nodes:INode[]
+        const routePathLinks:IRoutePathLinkResult[]
         = suunta.reitinlinkkisByReitunnusAndSuuvoimastAndSuusuunta.edges
-            .map((node: any) => NodeFactory.createNode(internalRoutePathId, node.node));
+            .map((node: any) =>
```
Comment (neodyymi) [inline src/factories/routePathFactory.ts]: Reitinlinkki = node? Node != solmu?

File: src/factories/routePathLinkFactory.ts (hunk 5 above)
Comment (neodyymi) [reply src/factories/routePathLinkFactory.ts]: Should be Link?

File: src/factories/routePathFactory.ts (hunk 6 above)
Comment (neodyymi) [reply src/factories/routePathFactory.ts]: Should be routePathLink or something

File: src/services/routeService.ts [hunk 7]
```diff
@@ -1,33 +1,67 @@
...
+    public static async fetchRoute(routeId: string): Promise<IRoute | undefined> {
+        return (await RouteService.runFetchRouteQuery(routeId)).route;
+    }
+
+    public static async fetchMultipleRoutes(routeIds: string[]):
+        Promise<IMultipleRoutesQueryResult> {
+        return new Promise<IMultipleRoutesQueryResult>((resolve, reject) => {
+            Promise
+                .all(routeIds.map(id => RouteService.runFetchRouteQuery(id)))
+                .then((queryResults) => {
```
Comment (neodyymi) [inline src/services/routeService.ts]: Use async/await

File: src/stores/nodeStore.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+    @computed get nodes(): INode[] {
+        return this._nodes;
+    }
+
+    set nodes(value: INode[]) {
+        this._nodes = value;
+    }
+
+    public getNodes(nodeIds: number[]) {
```
Comment (neodyymi) [inline src/stores/nodeStore.ts]: Why not just do `this.props.nodeStore!.nodes.filter(.....` in whatever is calling this? Or at least change the name to be more descriptive

File: src/stores/nodeStore.ts [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @action
+    public addToNodes(nodes: INode[]) {
+        this._nodes.push(...nodes);
+    }
+
+    @action
+    public removeFromNodes(nodeIds: number[]) {
+        this._nodes.forEach((node, index) => {
```
Comment (neodyymi) [inline src/stores/nodeStore.ts]: filter might be more suitable for this

Comment (neodyymi) [reply src/stores/nodeStore.ts]: `filter( node => !nodeIds.includes(node.id))`

File: src/stores/nodeStore.ts (hunk 8 above)
Comment (neodyymi) [reply src/stores/nodeStore.ts]: Probably not used

**Comment by kronken** [comment]
Fixd, can be reviewed again

**Pull request merged**